### PR TITLE
Add Beok TGR81 WiFi Controller/Thermostat

### PR DESCRIPTION
**Pull request by AlfredBroda**
Most functionality is done, I still have some issues with HA sensors and (hardware) sensor selector.

### CODE REVIEW

**Comment by AlfredBroda** [comment]
Everything works! I just had to learn proper sensor semantics. :)

File: custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            max: 20
+    - id: 101
+      type: integer
+      name: unknown_101
+    - id: 102
+      type: string
+      name: unknown_102
+
+secondary_entities:
+  - entity: sensor
```
Comment (make-all) [inline custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml:65]: With other devices, I have not duplicated the current temperature as a sensor, as it is available from the climate device and if you really want it as a separate sensor, you can use a template sensor to extract it.

File: custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      type: integer
+      name: min_temperature
+      optional: true
+      mapping:
+        - scale: 1
+          step: 1
+          range:
+            min: 5
+            max: 20
+    - id: 101
```
[the remark this answers is not in the document]
Comment (AlfredBroda) [reply custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml:57]: There is mention of a status for "factory_reset" just after "valve_state" and before "child_lock", but it does not seem to be represented in the DPS. I also wasn't able to change values fo 101 or 102 in any way through the original app.
```
{"devId": "24746323e09806aaab41", "dps": {"1": true, "2": "manual", "10": true, "16": 20, "19": 40, "24": 230, "27": 0, "36": "close", "40": false, "43": "in", "45": 0, "101": 1, "102": "70", "updated_at": 1663252327.3686218}}
```

File: custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          max: 9
+  - entity: switch
+    name: Anti-frost
+    icon: "mdi:snowflake-melt"
+    category: config
+    dps:
+      - id: 10
+        type: boolean
+        name: switch
+  - entity: sensor
```
[the remark this answers is not in the document]
Comment (AlfredBroda) [reply custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml]: Nice! I didn't know the functions could be separate - works like a charm.

File: custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml (hunk 1 above)
Comment (AlfredBroda) [reply custom_components/tuya_local/devices/beok_tgr81_thermostat_c.yaml]: Removed.

**Comment by AlfredBroda** [comment]
I made all of the proposed changes and fixed an issue that popped up when the device was removed and readded to the Tuya network. this is the final thing. :)

**Pull request merged**